### ISSUE TITLE
Explore Sensitivity of IGO Window Sizes on VBET Metrics

### ISSUE THREAD

**Issue by joewheaton** [Riverscapes/riverscapes-tools#684]
You can take any of the DGO attribute tables and assemble VBET metrics for example at different windows sizes and produce plots like this from Gary's paper:
![image](https://user-images.githubusercontent.com/9723905/207916782-183c7ad0-474c-4fc9-99c6-fef9bc599ae9.png)

Of course you can map the differences to:
![image](https://user-images.githubusercontent.com/9723905/207916939-b81fca91-97ee-4e80-af29-5aec773e36de.png)

_Originally posted by @joewheaton in https://github.com/Riverscapes/riverscapes-tools/issues/679#issuecomment-1353369864_

**Comment by lauren-herbine** [comment]
@joewheaton @shelbysawyer I am having a little difficulty in wrapping my head around how to do this efficiently:

I can't just multiply the current metric by something (i.e. current metric is at 100m, multiply by 2 to get a metric at 200 m), as that is most certainly not actually measuring the, say, active channel area over that 200 m window.

What I am struggling with is how I can do a sensitivity analysis at scale (more than just one or two riverscape reaches). If I were doing this for just one or two, I could manually measure the metric at each of the 14 different window sizes. That wouldn't really give me a good idea beyond those one or two reaches. To get a more accurate sense of how these different window sizes affect the metric accuracy, and to be able to sort by 0, 1, and 2 streams, I'd need to do this on in at least one HUC 8, ideally multiple. As I can't tell (code) VBET/GNAT to do give me this sort of output, do either of you have any ideas or advice? Maybe I'm totally missing something obvious here.

Additionally, are there any projects with the mainstems run? That would be very important to include in the analysis.

Thanks!

**Comment by lauren-herbine** [comment]
As I'm continuing to explore how to do this:

* we'd have to calculate metrics at 42 different window sizes per DGO (0, 1, 2 size streams at 14 analysis window sizes)
* say we did sensitivity analysis for 5 metrics- that would be 210 calculations per DGO... to do that at a HUC or multiple HUC scale would be massive...
* we could only do smaller sizes for 0 size streams (say- 200, 400), mid range ones (say- 400, 600, 800, 1000, 1200) for size 1 streams, and large (say- 1000 and up) for size 2 streams. This would still be many calculations and would be pretty complicated
* We don't have necessary intermediates _in the DGO/IGO format_ in attribute tables
 - [ ]  Channel lengths
 - [ ] elevations at top and bottom of DGOs
 - [ ] straight line distances
 - [ ] etc.

![image](https://user-images.githubusercontent.com/73319684/207951859-43886fe9-4858-4365-8ff1-925e523ff8c1.png)



|                                                   |                  |             |       |         |       |         |        |                                     |                                                              |                 Using Projects with 50   m DGO spacing                 |                                                                                                                                                                                                                                         |                                                                        |                                                                                                                                                                                                                                         |      |
|---------------------------------------------------|------------------|-------------|-------|---------|-------|---------|--------|-------------------------------------|--------------------------------------------------------------|:----------------------------------------------------------------------:|:---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------:|:----------------------------------------------------------------------:|:---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------:|:----:|
|                                                   |                  | Stream Size |       |         |       |         |        |                 LOE                 |                          Calculation                         |                               New Window                               |                                                                                                                                                                                                                                         |                                                                        |                                                                                                                                                                                                                                         |      |
|                                                   | Metric           |      0      |       |    1    |       |    2    |        |                                     |                                                              |                                   200                                  |                                                                                                                                                                                                                                         |                                   400                                  |                                                                                                                                                                                                                                         | ETC. |
|                                                   |                  |   current   |  new  | current |  new  | current |   new  |                                     |                                                              |                                  Look?                                 |                                                                                                                Calculate                                                                                                                |                                  Look?                                 |                                                                                                                Calculate                                                                                                                |      |
|                         ?                         | Stream Size      | 50 m        | 100 m | 50 m    | 100 m | 50 m    | 100 m  | drainage area + NHD   Fcode         | bankful script                                               |                                                                        |                                                                                                                                                                                                                                         |                                                                        |                                                                                                                                                                                                                                         |      |
|                    Areas (add)                    | Channel Area     | 100 m       | 200 m | 250 m   | 600 m | 1000 m  | 1000 m | CA polygon                          | Area w/in window                                             | look upstream 100   meters (2 IGOs) and downstream 100 meters (2 IGOs) | add the areas for each   IGO (4 total) together, assign to DGO                                                                                                                                                                          | look upstream 200   meters (4 IGOs) and downstream 200 meters (4 IGOs) | add the areas for each   IGO (8 total) together, assign to DGO                                                                                                                                                                          |      |
|                                                   | Floodplain Area  | 100 m       | 200 m | 250 m   | 600 m | 1000 m  | 1000 m | FLDPLN polygon                      | Area w/in window                                             | look upstream 100   meters (2 IGOs) and downstream 100 meters (2 IGOs) |                                                                                                                                                                                                                                         | look upstream 200   meters (4 IGOs) and downstream 200 meters (4 IGOs) |                                                                                                                                                                                                                                         |      |
| Length ratios (add lenghts and recalculate ratio) | Integrated Width | 100 m       | 200 m | 250 m   | 600 m | 1000 m  | 1000 m |                                     |                                                              | look upstream 100   meters (2 IGOs) and downstream 100 meters (2 IGOs) |                                                                                                                                                                                                                                         | look upstream 200   meters (4 IGOs) and downstream 200 meters (4 IGOs) |                                                                                                                                                                                                                                         |      |
|                                                   | Sinuosity        | 100 m       | 200 m | 250 m   | 600 m | 1000 m  | 1000 m | Stream network +   centerlines      | network   length/straight line distance                      | look upstream 100   meters (2 IGOs) and downstream 100 meters (2 IGOs) | add together the   channel lengths of the 4 IGOS, add together the centerline lengths of the 4   IGOs, perform new ratio calculation- assign to DGO                                                                                     | look upstream 200   meters (4 IGOs) and downstream 200 meters (4 IGOs) | add together the   channel lengths of the 8 IGOS, add together the centerline lengths of the 8   IGOs, perform new ratio calculation- assign to DGO                                                                                     |      |
|                                                   | Channel Gradient | 100 m       | 200 m | 250 m   | 600 m | 1000 m  | 1000 m | Stream network + DEM   (elevations) | (top of reach elev -   bottom of reach elev)/channel length) | look upstream 100   meters (2 IGOs) and downstream 100 meters (2 IGOs) | take top of reach elev   from top of furthest upstream IGO, take bottom of reach elev from bottom of   furthest downstream IGO. Find difference. Add centerline lengths of all IGOs   (4). Perform new ratio calculation- assign to DGO | look upstream 200   meters (4 IGOs) and downstream 200 meters (4 IGOs) | take top of reach elev   from top of furthest upstream IGO, take bottom of reach elev from bottom of   furthest downstream IGO. Find difference. Add centerline lengths of all IGOs   (8). Perform new ratio calculation- assign to DGO |      |

**Comment by lauren-herbine** [comment]
Yes! I'll ping you!

**Comment by joewheaton** [comment]
Thanks for this work @lauren-herbine. Can you work with @shelbysawyer and transfer this over to her, and have her repeat the analysis you did, with our new IGO spacing. 

@shelbysawyer the outcome of this is to decide if we are happy with or need to change the window sizes for our IGO metric windows. I had @lauren-herbine work on it because it is central to her work on GNAT and you were busy. Please get the final updates to @jtgilbert before we run 1.0 (i.e. in 0.9.1 to 0.9.X to check you like them.

**Comment by lauren-herbine** [comment]
@joewheaton yes, we can work on it! I'll be getting the updated GNAT today (hopefully), so we can both work through this on our respective projects
